backtest/engine.py

```
@@ -0,0 +1,427 @@
+        if max_up >= exit_thr:
+            reversal_seen_down = True
+
+        # --- PAIR COMPLETION ---
+        if filled_up and filled_down and not pair_captured:
+            pair_captured = True
+            pnl_cents += 4.0
+            pnl_cents -= (params.merge_gas_usd * 100.0 /
+                          max(1, params.quote_shares // params.min_quote_shares))
+
+    if filled_up or filled_down:
+        if (filled_up and not filled_down) or (filled_down and not filled_up):
```

Ruff (0.16.2)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)